docs/source/custom_resolvers.rst

```diff
@@ -0,0 +1,292 @@
+    ...         "uncached": "${uncached:0,10000}",
+    ...         "cached_1": "${cached:0,10000}",
+    ...         "cached_2": "${cached:0, 10000}",
+    ...         "cached_3": "${cached:0,${uncached}}",
+    ...     }
+    ... )
+    >>> # not the same since the cache is disabled by default
+    >>> assert c.uncached != c.uncached
+    >>> # same value on repeated access thanks to the cache
+    >>> assert c.cached_1 == c.cached_1 == 122
+    >>> # same input as `cached_1` => same value
+    >>> assert c.cached_2 == c.cached_1 == 122
+    >>> # same string literal "${uncached}" => same value
+    >>> assert c.cached_3 == c.cached_3 == 1192
+
+
+Custom interpolations can also receive the following special parameters:
+
+- ``_parent_`` : the parent node of an interpolation.
+- ``_root_``: The config root.
+
+This can be achieved by adding the special parameters to the resolver signature.
+Note that special parameters must be defined as named keywords (after the `*`):
+
+In this example, we use ``_parent_`` to implement a sum function that defaults to 0 if the node does not exist.
+(In contrast to the sum we defined earlier where accessing an invalid key, e.g. ``"a_plus_z": ${sum:${a}, ${z}}`` will result in an error).
+
+.. doctest::
+
+    >>> def sum2(a, b, *, _parent_):
+    ...     return _parent_.get(a, 0) + _parent_.get(b, 0)
+    >>> OmegaConf.register_new_resolver("sum2", sum2, use_cache=False)
+    >>> cfg = OmegaConf.create(
+    ...     {
+    ...         "node": {
+    ...             "a": 1,
+    ...             "b": 2,
+    ...             "a_plus_b": "${sum2:a,b}",
+    ...             "a_plus_z": "${sum2:a,z}",
+    ...         },
+    ...     }
+    ... )
+    >>> cfg.node.a_plus_b
+    3
+    >>> cfg.node.a_plus_z
+    1
+
+
+Built-in resolvers
+------------------
+
+.. _oc.env:
+
+oc.env 
+^^^^^^
+
+Access to environment variables is supported using ``oc.env``:
+
+Input YAML file:
+
+.. include:: env_interpolation.yaml
+   :code: yaml
+
+.. doctest::
+
+    >>> conf = OmegaConf.load('source/env_interpolation.yaml')
+    >>> conf.user.name
+    'omry'
+    >>> conf.user.home
+    '/home/omry'
+
+You can specify a default value to use in case the environment variable is not set.
+In such a case, the default value is converted to a string using ``str(default)``, unless it is ``null`` (representing Python ``None``) - in which case ``None`` is returned. 
+
+The following example falls back to default passwords when ``DB_PASSWORD`` is not defined:
+
+
+.. _oc.deprecated:
+
+oc.deprecated
+^^^^^^^^^^^^^
+``oc.deprecated`` enables you to deprecate a config node.
+It takes two parameter:
+- key: An interpolation key representing the new key you are migrating to. This parameter is required.
+- message: A message to use as the warning when the config node is being accessed. The default message is
+``'$KEY' is deprecated. Change your code and config to use '$NEW_KEY'``. Note that $KEY and $NEW_KEY 
+do not use interpolation syntax.
```

Comment:
```suggestion
It takes two parameters:

- ``key``: An interpolation key representing the new key you are migrating to. This parameter is required.
- ``message``: An optional message to use as the warning when the config node is being accessed. The default message is
  ``'$KEY' is deprecated. Change your code and config to use '$NEW_KEY'``. Note that ``$KEY`` and ``$NEW_KEY``
  do not use interpolation syntax.
```

Reply:
assuming this is only about using \`\` around key and message, done.

Reply:
It was more than that => I created another diff